src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,468 @@
+                            selectedFeature.getFeatureTable().updateFeatureAsync(selectedFeature);
+
+                            // refresh ui to enable syncing
+                            currentEditState = EditState.READY;
+                            geodatabaseButton.setText("Sync Geodatabase");
+                            geodatabaseButton.setDisable(false);
+                          }
+                        });
+                      }
+                    } catch (InterruptedException | ExecutionException e) {
+                      displayMessage("Exception getting selected feature", e.getMessage());
+                    }
+                  });
+                }
+              } catch (InterruptedException | ExecutionException e) {
+                displayMessage("Exception getting clicked feature", e.getMessage());
+              }
+            });
+          });
+
+        } else if (event.getButton() == MouseButton.SECONDARY) {
+
+          // on secondary mouse click, clear feature selection
+          for (Layer layer : mapView.getMap().getOperationalLayers()) {
+            FeatureLayer featureLayer = (FeatureLayer) layer;
+            featureLayer.clearSelection();
+          }
+        }
+      });
+
+      // add map view, controlsVBox and progressBar to stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox, progressIndicator);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+
+    } catch (Exception e) {
+      // on any error, display the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Generates a local geodatabase and sets it to the map.
+   */
+  private void generateGeodatabase() {
+
+    // define geodatabase sync task
+    geodatabaseSyncTask = new GeodatabaseSyncTask("https://sampleserver6.arcgisonline.com/arcgis/rest/services/Sync/WildfireSync/FeatureServer");
+    geodatabaseSyncTask.loadAsync();
+    geodatabaseSyncTask.addDoneLoadingListener(() -> {
+
+      // create generate geodatabase parameters for the current extent
+      final ListenableFuture<GenerateGeodatabaseParameters> defaultParametersFuture = geodatabaseSyncTask
+              .createDefaultGenerateGeodatabaseParametersAsync(geodatabaseExtentGraphics.getGeometry());
+      defaultParametersFuture.addDoneListener(() -> {
+        try {
+          // set parameters and don't include attachments
+          GenerateGeodatabaseParameters defaultParameters = defaultParametersFuture.get();
+          defaultParameters.setReturnAttachments(false);
+
+          // create a temporary file for the geodatabase
+          File tempFile = File.createTempFile("gdb",".geodatabase");
+          tempFile.deleteOnExit();
+
+          // create and start the job
+          GenerateGeodatabaseJob geodatabaseJob = geodatabaseSyncTask.generateGeodatabase(defaultParameters, tempFile.getAbsolutePath());
+          geodatabaseJob.start();
```

Comment:
I used the progress bar originally, but because the progress hovers at ca 5% for a long time and then jumps to 100%, I found it nicer to have the progress indicator to reassure the user the sample isn't frozen. Can I suggest both?

Reply:
I think the jumping progress is fixed in 100.6 so it might be an idea to go with the progress bar knowing it will come good in the next release.